packages/protocol/contracts/common/GasPriceMinimum.sol

```diff
@@ -1,43 +1,44 @@
 pragma solidity ^0.5.8;
 
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
 import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
 import "./Initializable.sol";
 import "./UsingRegistry.sol";
+import "./UsingFixidity.sol";
 import "../stability/interfaces/ISortedOracles.sol";
-import "../common/FractionUtil.sol";
 
 /**
  * @title Stores and provides gas price minimum for various currencies.
  */
-contract GasPriceMinimum is Ownable, Initializable, UsingRegistry {
+contract GasPriceMinimum is Ownable, Initializable, UsingRegistry, UsingFixidity {
   using SafeMath for uint256;
-  using FractionUtil for FractionUtil.Fraction;
 
   event TargetDensitySet(
-    uint256 numerator,
-    uint256 denominator
+    int256 targetDensity
   );
 
   event AdjustmentSpeedSet(
-    uint256 numerator,
-    uint256 denominator
+    int256 adjustmentSpeed
   );
 
   event InfrastructureFractionSet(
-    uint256 numerator,
-    uint256 denominator
+    int256 infrastructureFraction
   );
 
   uint256 public gasPriceMinimum;
 
   // Block congestion level targeted by the gas price minimum calculation.
-  FractionUtil.Fraction public targetDensity;
+  int256 public targetDensity;
```

Comment:
I wonder if it makes sense to wrap `int256` into a `Fraction` type and make it more clear.

Similar to what openzeppelin does with `Counter` [here](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/drafts/Counters.sol)

Reply:
Yep, that's something I've considered. Since this PR is already huge, I think the best way forward is to put this in a follow-up PR (there's some other things a follow-up PR would include).

Reply:
mixed feeling about that...
I understand the need to close this and continue, but at the same time, is probably something that will remain as tech debt, and probably is harder to track later than to track it today. I would at least check how easy is to do that.

I also understand that in terms of gas optimization is not great...  Maybe the "best" solution would be to change fixidity to have a type, similar to what Counter does.